src/figuresClasses.ts

```diff
@@ -1,11 +1,75 @@
-export interface Figure {}
+type Shape = `triangle` | `circle` | `rectangle`;
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides: number[] = [this.a, this.b, this.c];
+    const maxSide = Math.max(...sides);
+    const sumOfTwoSides = sides.reduce((sum, side) => sum + side, 0) - maxSide;
+
+    if (sides.some((side) => side <= 0) || sumOfTwoSides <= maxSide) {
+      throw new Error('invalid data for figure, all side must be > 0 and the biggest one less than sum of other two');
```

Comment:
The error message could be clearer. Consider rephrasing it to: 'Invalid triangle sides: all sides must be > 0, and the largest side must be less than the sum of the other two.'